pimd/pim6_cmd.c

```diff
@@ -448,6 +448,448 @@ DEFPY (no_ipv6_pim_rp_prefix_list,
 	return pim_process_no_rp_plist_cmd(vty, rp_str, plist);
 }
 
+DEFPY (show_ipv6_pim_rp,
+       show_ipv6_pim_rp_cmd,
+       "show ipv6 pim [vrf NAME] rp-info [X:X::X:X/M$group] [json$json]",
+       SHOW_STR
+       IPV6_STR
+       PIM_STR
+       VRF_CMD_HELP_STR
+       "PIM RP information\n"
+       "Multicast Group range\n"
+       JSON_STR)
+{
+	struct pim_instance *pim;
+	struct vrf *v;
+	json_object *json_parent = NULL;
+	struct prefix *range = NULL;
+
+	v = vrf_lookup_by_name(vrf ? vrf : VRF_DEFAULT_NAME);
+
+	if (!v)
+		return CMD_WARNING;
+
+	pim = pim_get_pim_instance(v->vrf_id);
+
+	if (!pim) {
+		vty_out(vty, "%% Unable to find pim instance\n");
+		return CMD_WARNING;
+	}
+
+	if (group_str) {
+		range = prefix_new();
```

Comment:
This is leaking memory, the prefix allocated here is never freed.

Instead of using `prefix_new()`, just use `struct prefix range;` (no pointer)